desktop/libs/notebook/src/notebook/connectors/trino.py

```diff
@@ -321,6 +321,24 @@ def _get_columns(self, database, table):
     ]
 
 
+  @query_error_handler
+  def explain(self, notebook, snippet):
+    statement = snippet['statement'].rstrip(';')
+    explanation = ''
+
+    if statement:
+      query_client = TrinoQuery(self.trino_request, 'USE ' + snippet['database'])
+      query_client.execute()
+      query_client = TrinoQuery(self.trino_request, 'EXPLAIN ' + statement)
+      result = query_client.execute() 
+
+    return {
+      'status': 0,
```

Comment:
Will the TrinoQuery.execute() throws any exception?
If any exception, should the status be updated to -1?
Take a look existing beeswax API examples: response = {'status': -1, 'message': ''} are defined at the beginning of menthod.

Reply:
Added the try catch.
We need to send the {'status': 0} to capture the output in UI. 
For the ref https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/hiveserver2.py#L591

Reply:
@wing2fly we should try to move away from using the status flag, today API users expect the proper http error code when something goes bad. That being said the UI might have some old code that checks for status == 0.